From aee9cd2b8db137be3574984fd036f1556cf34b27 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Tue, 13 Feb 2018 09:33:44 -0800
Subject: [PATCH 0273/1852] Revert "drm: xlnx: zynqmp: disp: Use the device
 node for layers"

commit b24a79a8cba5e493d1617f7c2f8750f75a3d8563 from
https://github.com/Xilinx/linux-xlnx.git

This reverts commit d1cd1f25e305e612873c3e1e4637c07262d8e1cf.

Without Xilinx bridge support, layers don't need own child nodes.
But if a layer is exposed as a Xilin bridge, it needs to have
a dedicated node to be referenced by other client.

The change will have to submitted along with Xilinx bridge
patch set.

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/zynqmp_disp.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/xlnx/zynqmp_disp.c b/drivers/gpu/drm/xlnx/zynqmp_disp.c
index f46be97d3fc8..5f527671b88b 100644
--- a/drivers/gpu/drm/xlnx/zynqmp_disp.c
+++ b/drivers/gpu/drm/xlnx/zynqmp_disp.c
@@ -2014,12 +2014,17 @@ static int zynqmp_disp_layer_create(struct zynqmp_disp *disp)
 	int ret;
 
 	for (i = 0; i < ZYNQMP_DISP_NUM_LAYERS; i++) {
+		char temp[16];
+
 		layer = &disp->layers[i];
 		layer->id = i;
 		layer->offset = i * 4;
 		layer->other = &disp->layers[!i];
 		layer->num_chan = num_chans[i];
-		layer->of_node = disp->dev->of_node;
+		snprintf(temp, sizeof(temp), "%s-layer", dma_name[i]);
+		layer->of_node = of_get_child_by_name(disp->dev->of_node, temp);
+		if (!layer->of_node)
+			goto err;
 		ret = zynqmp_disp_layer_request_dma(disp, layer, dma_name[i]);
 		if (ret)
 			goto err;
-- 
2.31.1

